File: talleres/taller01/Date/Date.py
class Date:
    def __init__(self, day, month, year):
        self.day = day
        self.month = month
        self.year = year

    def get_day(self):
        return self.day

    def get_month(self):
        return self.month

    def get_year(self):
        return self.year

    def compare_dates(self, other_date):
        if self.year < other_date.year or self.year == other_date.year and self.month < other_date.month or self.year == other_date.year and self.month == other_date.month and self.day < other_date.day:
            print("La fecha actual es antes que la comparada")
        elif self.year > other_date.year or self.year == other_date.year and self.month > other_date.month or self.year == other_date.year and self.month == other_date.month and self.day > other_date.day:
            print("La fecha comparada es antes que la actual")
        else:
            print("Las fechas son iguales")

    def __str__(self):
        return ("Date: "+str(self.day)+"/"+str(self.month)+"/"+str(self.year))

File: talleres/taller01/Date/Date.cpp
#include <iostream>

using namespace std;

class Date
{
    int day;
    int month;
    int year;

public:
    Date(int day, int month, int year){
        this->day = day;
        this->month = month;
        this->year = year;
    }

    int get_day(){
        return this->day;
    }

    int get_month(){
        return this->month;
    }

    int get_year(){
        return this->year;
    }

    int compare_dates(Date *other){
        if (this->year < other->year or this->year == other->year and this->month < other->month or this->year == other->year and this->month == other->month and this->day < other->day){
            cout << "-1" << endl;
        }
        else if (this->year > other->year or this->year == other->year and this->month > other->month or this->year == other->year and this->month == other->month and this->day > other->day){
            cout << "0" << endl;
        }
        else{
            cout << "1" << endl;
        }
    }
};

std::ostream &operator<<(std::ostream &out, Date &date){
    return out << "Date: " + to_string(date.get_day()) + "/" + to_string(date.get_month()) + "/" + to_string(date.get_year()) << endl;
}

int main(){
    Date hoy(27, 1, 2021), ayer(26, 1, 2021);

    cout << hoy.get_day() << endl;
    hoy.compare_dates(&ayer);
    cout << hoy << endl;
}